rules/how-to-read-source-code/rule.md

```diff
@@ -0,0 +1,40 @@
+---
+type: rule
+title: Do you know how to read sourse code?
+uri: do-you-know-how-to-read-source-code
+authors:
+  - title: Luke Mao
+    url: https://www.ssw.com.au/people/luke-mao
+  - title: Andrew Harris
+    url: https://www.ssw.com.au/people/andrew-harris
+  - title: Matt Goldman
+    url: https://www.ssw.com.au/people/matt-goldman
+created: 2022-09-12T14:44:28.666Z
+guid: 1b706f91-a99c-49a6-9ae8-72d2026abc11
+---
+First of all, you need to have the following prerequisites so that you can read the code smoothly afterwards.
+
+1. **Basic knowledge** - Knowledge of relevant languages ​​and underlying technologies.
+2. **Software function** - You must first know what kind of functions this software completes, what features it has, and what configurations it has. You need to read the user manual first, then let the software run, and feel it for yourself.
+3. **Relevant documentation** - Read the relevant internal documents, Readme or Release Notes, Design or Wiki. These documents can let you understand all aspects of the software. If your software doesn't have documentation, then you can only count on the original author of the software still alive and willing to communicate.
+4. **The structure of the code** - You need to know what is the function of each directory. If the program you want to read is organized under some standard framework, such as the [Clean Architecture](https://github.com/jasontaylordev/CleanArchitecture). Then congratulations, the code is not difficult to read.
+
+Next, you need to understand what parts of the code of this software are made up of. Below is a list for reference.
+
+1. **Interface/abstract definition** - Any code will have many interface or abstract definitions, which describe the data structures or business entities that the code needs to deal with and the relationships between them. It is very important to understand these relationships.
```

Comment:
>interface

should be interfaces?

Also:
> which describe the data structures or business entities that the code needs

Should be:
which describe the functionality that the code needs

Reply:
> which describe the functionality that the code needs

Here it focuses on understanding the relationships between the data entities. Functionality is how we manipulate them. So I prefer the original one.